src/subscribers/longpoll.c

```diff
@@ -524,6 +541,42 @@ static void request_cleanup_handler(subscriber_t *sub) {
 }
 
 
+static void keep_alive_handler(ngx_event_t *ev) {
+  full_subscriber_t *fsub = (full_subscriber_t *)ev->data;
+
+  if(ev->timedout) {
+    ngx_buf_t   *buf = REQUEST_PCALLOC(fsub->sub.request, buf);
+    ngx_chain_t *chain = REQUEST_PALLOC(fsub->sub.request, chain);
+    nchan_loc_conf_t       *cf = ngx_http_get_module_loc_conf(fsub->sub.request, ngx_nchan_module);
+    size_t                  separator_len = cf->subscriber_http_raw_stream_separator.len;
+
+    ev->timedout=0;
+    ngx_add_timer(&fsub->data.keep_alive_ev, fsub->sub.cf->subscriber_keep_alive_interval * 1000);
+
+    chain->buf=buf;
+    chain->next=NULL;
+
+    buf = chain->buf;
+    ngx_memzero(buf, sizeof(ngx_buf_t));
+    buf->start = cf->subscriber_http_raw_stream_separator.data;
+    buf->pos = buf->start;
+    buf->end = buf->start + separator_len;
+    buf->last = buf->end;
+
+    buf->last_in_chain = 1;
+    buf->memory = 1;
+    buf->last_buf = 1;
+    buf->flush = 1;
+
+    if(!fsub->data.shook_hands) {
+      nchan_cleverly_output_headers_only_for_later_response(fsub->sub.request);
+      fsub->data.shook_hands = 1;
+    }
+
+    nchan_output_filter(fsub->sub.request, chain);
```

Comment:
I've figured out where the \n was coming from, the http raw separator id automatically adding them